aws_doc_sdk_examples_tools/agent/make_prompts.py

```diff
@@ -20,12 +20,17 @@ def make_doc_gen(root: Path) -> DocGen:
     return doc_gen
 
 
-def write_prompts(snippets: Dict[str, Snippet], out: Path) -> None:
-    """Write each snippet's code into a separate Markdown file."""
+def write_prompts(doc_gen: DocGen, out: Path) -> None:
     out.mkdir(parents=True, exist_ok=True)
-    for snippet_id, snippet in snippets.items():
-        snippet_path = out / f"{snippet_id}.md"
-        snippet_path.write_text(snippet.code, encoding="utf-8")
+    examples = doc_gen.examples
+    snippets = doc_gen.snippets
+    for example_id, example in examples.items():
+        # Postfix with `.md` so Ailly will pick it up.
+        prompt_path = out / f"{example_id}.md"
+        # This assumes we're running DocGen specifically on AWSIAMPolicyExampleReservoir.
```

Comment:
Is there a place we can add this as an assertion? Like at... line 69? `assert 'AWSIAMPolicyExampleReservoir' in doc_gen_path`?... er, in whatever the tributary path is...